File: frontend/src/types/letter.ts
import type { Locale } from '../i18n/types';

export interface LetterContent {
  title: string;
  category: string;
  description: string;
  content: string;
}

export interface Letter {
  id: number;
  title: string;
  category: string;
  description: string;
  image: string;
  content: string;
  createdLocale?: Locale;
  translations?: Partial<Record<Locale, LetterContent>>;
}

export interface LetterListResponse {
  letters: Letter[];
  total: number;
  page: number;
  page_size: number;
  categories: string[];
}

File: backend/app/schemas/letter.py
from pydantic import BaseModel, Field


class Letter(BaseModel):
    id: int
    title: str
    category: str
    description: str
    image: str
    content: str
    user_id: str | None = None


class LetterListResponse(BaseModel):
    letters: list[Letter]
    total: int
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=12, ge=1)
    categories: list[str]
